src/mapping/nav2_map_server/src/map_reps/occupancy_grid.cpp

```diff
@@ -0,0 +1,353 @@
+/*
+ * Copyright (c) 2008, Willow Garage, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Willow Garage, Inc. nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * This file contains helper functions for loading images as maps.
+ *
+ * Author: Brian Gerkey
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <libgen.h>
+#include <fstream>
+#include <cstring>
+#include <stdexcept>
+#include "yaml-cpp/yaml.h"
+#include "rclcpp/rclcpp.hpp"
+#include "map_server/map_reps/occupancy_grid.h"
+#include "nav_msgs/srv/get_map.hpp"
+#include "nav_msgs/msg/map_meta_data.hpp"
+#include <SDL/SDL_image.h>
+#include <LinearMath/btQuaternion.h>
+
+#define MAP_IDX(sx, i, j) ((sx) * (j) + (i))
+
+#ifdef HAVE_YAMLCPP_GT_0_5_0
```

Comment:
This was required in the old map_server for backwards support.

I think ROS2 targets all include yaml GT 0.5 and backwards support can now be dropped.

Reply:
Thanks for pointing this out. Dropped backwards support in latest commit